GANDLF/metrics/classification.py

```diff
@@ -62,46 +64,27 @@ def overall_stats(prediction: torch.Tensor, target: torch.Tensor, params: dict)
                 num_classes=params["model"]["num_classes"],
                 average=average_type_key,
             ),
-            # "aucroc": tm.AUROC(
-            #     task=task,
-            #     num_classes=params["model"]["num_classes"],
-            #     average=average_type_key if average_type_key != "micro" else "macro",
-            # ),
+            "auroc": tm.AUROC(
+                task=task,
+                num_classes=params["model"]["num_classes"],
+                average=average_type_key if average_type_key != "micro" else "macro",
+            ),
         }
         for metric_name, calculator in calculators.items():
-            if metric_name == "aucroc":
-                one_hot_preds = one_hot(
-                    prediction.long(), num_classes=params["model"]["num_classes"]
-                )
+            # TODO: AUROC needs to be properly debugged for multi-class problems - https://github.com/mlcommons/GaNDLF/issues/817
+            if metric_name == "auroc" and params["model"]["num_classes"] == 2:
```

Comment:
Don't really get the logic of the if-elif statement; for both conditions we are calling get_output_from_calculator the same way, no? If so, then why adding the conditional statement?

Reply:
I think `auroc` is broken for multi-class classification (#817). I am not 100% sure what the issue is since I have not gone deep into it, but I just wanted to put a guardrail in place for this specific PR.

Reply:
If so then let's leave it that way, but myself I have successfully used AUROC from torchmetrics in one of projects where I had multiclass classification, it worked just fine. Needs ingvestigation!

Reply:
Please provide details in that issue. Thanks!